apps/andi/lib/andi/input_schemas/ingestion.ex

```diff
@@ -181,6 +182,30 @@ defmodule Andi.InputSchemas.Ingestion do
     {extract_step_changesets, extract_step_errors}
   end
 
+  def merge_transformation_changeset(%Ecto.Changeset{data: %Andi.InputSchemas.Ingestion{}} = ingestion_changeset, transformation_changesets) do
+    transformation_changeset_list =
+      Enum.reduce(transformation_changesets, [], fn transformation_changeset, acc ->
+        transformation_changes = StructTools.to_map(Changeset.apply_changes(transformation_changeset))
+        [transformation_changes | acc]
+      end)
+
+    cleared_ingestion_changeset = Changeset.delete_change(ingestion_changeset, :transformations)
+
+    changeset(cleared_ingestion_changeset, %{transformations: transformation_changeset_list})
+  end
+
+  def get_transformation_changesets(ingestion_changeset) do
+    transformations =
+      case Changeset.fetch_change(ingestion_changeset, :transformations) do
+        {_, transformations} -> transformations
+        :error -> []
+      end
+
+    Enum.reduce(transformations, [], fn transformation, acc ->
+      acc ++ [Transformation.validate(transformation)]
```

Comment:
Ah, I think this is where I meant to leave my other comment. Could also keep the same format as the other spot using the | operator instead of ++